api/dbadapters/presto.ts

```diff
@@ -0,0 +1,228 @@
+// TODO: Move this to somewhere both the API and CLI can use?
+function resolveTarget(target: dataform.ITarget) {
+  return `${target.database ? `${target.database}.` : ""}${
+    target.schema ? `${target.schema}.` : ""
+  }${target.name}`;
+}
+
+export class PrestoDbAdapter implements IDbAdapter {
+  public static async create(
+    credentials: Credentials,
+    options?: { concurrencyLimit?: number; disableSslForTestsOnly?: boolean }
+  ) {
+    return new PrestoDbAdapter(credentials, options);
+  }
+
+  private prestoCredentials: dataform.IPresto;
+
+  private client: Presto.Client;
+
+  private constructor(credentials: Credentials, options?: { concurrencyLimit?: number }) {
+    this.prestoCredentials = credentials as dataform.IPresto;
+    this.client = new Presto.Client(this.prestoCredentials as Presto.IPrestoClientOptions);
+  }
+
+  public async execute(
+    statement: string,
+    // TODO: These execute options should actually allow any of Presto.IPrestoClientExecuteOptions.
+    options: {
+      rowLimit?: number;
+      byteLimit?: number;
+    } = { rowLimit: 1000, byteLimit: 1024 * 1024 }
+  ): Promise<IExecutionResult> {
+    const prestoResult = await prestoExecute(this.client, statement);
+    return { rows: prestoResult.data, metadata: {} };
+  }
+
+  public async withClientLock<T>(callback: (client: IDbClient) => Promise<T>) {
+    return await callback(this);
+  }
+
+  public async evaluate(queryOrAction: QueryOrAction, projectConfig?: dataform.ProjectConfig) {
+    const useSingleQueryPerAction =
+      projectConfig?.useSingleQueryPerAction === undefined ||
+      !!projectConfig?.useSingleQueryPerAction;
+    const validationQueries = collectEvaluationQueries(
+      queryOrAction,
+      useSingleQueryPerAction,
+      (query: string) => (!!query ? `explain ${query}` : "")
+    ).map((validationQuery, index) => ({ index, validationQuery }));
+    const validationQueriesWithoutWrappers = collectEvaluationQueries(
+      queryOrAction,
+      useSingleQueryPerAction
+    );
+
+    const queryEvaluations = new Array<dataform.IQueryEvaluation>();
+    for (const { index, validationQuery } of validationQueries) {
+      let evaluationResponse: dataform.IQueryEvaluation = {
+        status: dataform.QueryEvaluation.QueryEvaluationStatus.SUCCESS
+      };
+      try {
+        await prestoExecute(this.client, validationQuery.query);
+      } catch (e) {
+        evaluationResponse = {
+          // TODO: Parse the error with line number.
+          status: dataform.QueryEvaluation.QueryEvaluationStatus.FAILURE,
+          error: e
+        };
+      }
+      queryEvaluations.push(
+        dataform.QueryEvaluation.create({
+          ...evaluationResponse,
+          incremental: validationQuery.incremental,
+          query: validationQueriesWithoutWrappers[index].query
+        })
+      );
+    }
+    return queryEvaluations;
+  }
+
+  public async catalogs(): Promise<string[]> {
```

Comment:
should be `private`

Reply:
My thinking for having it public was for external users of `@dataform/api`, being able to query catalogs for Presto could be useful. What do you think?